src/main/java/org/helpberkeley/memberdata/RestaurantTemplateParser.java

```diff
@@ -196,16 +197,19 @@ private void processControlBlock() {
             }
         }
 
+        if (! hasFormulaDirective) {
+            throw new MemberDataException(
+                    TEMPLATE_ERROR + "\n" +
+                    "No Formula rows found within the Control Block. At least one valid Formula row is required.");
```

Comment:
Use a constant for this String. Update test to remove String duplication.

Also this pattern of TEMPLATE_ERROR + SOME_OTHER_CONSTANT looks like it could be improved by have multiple constants.  Example:
`
public static String TEMPLATE_ERROR = "Restaurant Template Error: "
public static String NO_FORMULA_ROWS_FOUND = TEMPLATE_ERROR + "No Formula rows found ...";
 `

Then use NO_FORMULA_ROWS_FOUND in the code and tests.  This probably also applies to other constants in the file.